uint8 predicate
string cmd
float64[] data
# bool need_device_response
# uint16 timeout_ms
---
bool success
string response       # raw reply or error message